### PR TITLE
fs: Remove NFFS support

### PR DESCRIPTION
**Pull request by nvlsianpu**
NFFS haven't been supported in its upstream since more than a year.
It has serous issues which causes it highly unreliable (see; https://github.com/apache/mynewt-nffs/issues/10 and others).

As <b>Litlefs</b> support is available in the zephyr since while it doesn't make any sense to keep NFFS support as it losted even education value: LitleFS is better substitute.

NFFS is removed "immediately" as [zephyr API deprecation process guide](https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#deprecated) stands:
> Unstable APIs can be removed without deprecation at any time. Deprecation and removal of APIs will be announced in the “API Changes” section of the release notes.

following projects were switched from NFFS to LittleFS
- tests/settings/functional/file test-case
- tests/lib/gui/lvgl test-case
- tests/subsys/settings/functional/file test-case
- samples/subsys/mgmt/mcumgr/smp_svr sample

Configuration of samples/subsys/shell/fs sample was truncated by NFFS related records.

tests/subsys/fs/multi-fs test-case was disabled. It will be reworked and restored to CI in another PR: #21973

### CODE REVIEW

File: samples/subsys/shell/fs/README.rst [hunk 1]
```diff
@@ -6,12 +6,12 @@ File system shell example
 Overview
 ********
 
-This example provides shell access to a NFFS file system partition in flash.
+This example provides shell access to a littlefs file system partition in flash.
```
Comment (de-nordic) [inline samples/subsys/shell/fs/README.rst]: Inconsistent naming of LittleFS within documentation; varies from littlefs to Littlefs to LittleFS

File: tests/subsys/fs/multi-fs/src/nffs_test_utils.c [hunk 2]
```diff
@@ -31,7 +31,7 @@ void nffs_test_util_overwrite_data(u8_t *data, u32_t data_len,
 	struct flash_pages_info info;
 	off_t off;
 
-	dev = device_get_binding(CONFIG_FS_NFFS_FLASH_DEV_NAME);
+	/*dev = device_get_binding();*/
```
Comment (de-nordic) [inline tests/subsys/fs/multi-fs/src/nffs_test_utils.c]: Are you sure that you want to leave the commented out code here?

File: tests/subsys/fs/multi-fs/src/test_nffs_mount.c [hunk 3]
```diff
@@ -29,7 +29,7 @@ static int test_mount(void)
 	struct device *flash_dev;
 	int res;
 
-	flash_dev = device_get_binding(CONFIG_FS_NFFS_FLASH_DEV_NAME);
+	/*flash_dev = device_get_binding();*/
```
Comment (de-nordic) [inline tests/subsys/fs/multi-fs/src/test_nffs_mount.c]: Are you sure that you want to leave the commented out code here?

File: doc/releases/release-notes-2.2.rst [hunk 4]
```diff
@@ -58,6 +58,10 @@ Removed APIs in this release
     SHELL_STATIC_SUBCMD_SET_CREATE
   * SHELL_CREATE_DYNAMIC_CMD (deprecated), replaced by SHELL_DYNAMIC_CMD_CREATE
 
+* Newtron Flash File System (NFFS) was removed. NFFS was removed sinc it has
```
Comment (carlescufi) [inline doc/releases/release-notes-2.2.rst]: s/sinc/since

**Review by carlescufi — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

File: doc/releases/release-notes-2.2.rst [hunk 5]
```diff
@@ -58,6 +58,10 @@ Removed APIs in this release
...
+    serious issues, not fixed since a long time. Wher it was possible
```
Comment (lemrey) [inline doc/releases/release-notes-2.2.rst]: s/Wher/Where

File: doc/releases/release-notes-2.2.rst [hunk 6]
```diff
@@ -58,6 +58,10 @@ Removed APIs in this release
...
+    NFFS usage were replaced by LittleFS usage as the better substitute.
```
Comment (lemrey) [inline doc/releases/release-notes-2.2.rst]: s/were/was

**Review by lemrey — COMMENTED** [review]
Something seems off with `tests/subsys/fs/multi-fs`, could you please double check that you have removed the nffs-related files there? Plus, which other FS will we be testing in that test, now that `nffs` was removed?

File: tests/subsys/settings/functional/file/native_posix.overlay [hunk 7]
```diff
@@ -19,7 +19,7 @@
 
 		storage_partition: partition@70000 {
 			label = "storage";
-			reg = <0x00070000 0x10000>;
```
Comment (lemrey) [inline tests/subsys/settings/functional/file/native_posix.overlay]: Why this change?

**Review by nvlsianpu — COMMENTED** [review]
@de-nordic 
> Inconsistent naming of LittleFS within documentation; varies from littlefs to Littlefs to LittleFS
use LittleFS in whole patch

@carlescufi 
Typos were fixed

> Something seems off with tests/subsys/fs/multi-fs, could you please double check that you have removed the nffs-related files there? Plus, which other FS will we be testing in that test, now that nffs was removed?

@lemrey 
Don't wary about this testsuite: It will be LittleFS - see #21973

**Review by lemrey — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]
